### PR TITLE
feat: frequency annotation helper

### PR DESCRIPTION
**Pull request by ADorigi**
## Issue
partially resolves https://github.com/validator-labs/validator/issues/358

## Description
- A constant was added to define the annotation key `validation.validator.labs/reconciliation-frequency`.
- Function `FrequencyFromAnnotations` which accepts annotation map and returns `ctrl.Result`.
- Tests for `FrequencyFromAnnotations` function.

### CODE REVIEW

**Comment by ADorigi** [comment]
No worries @TylerGillson  
I have updated the logs to reflect the changes.

File: pkg/plugins/plugin_frequency.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+// FrequencyFromAnnotations calculates reconciliation frequency from annotations of the plugin custom resource.
+// Defaults to 120 seconds if annotation is not found.
+func FrequencyFromAnnotations(l logr.Logger, annotations map[string]string) ctrl.Result {
+	var frequency time.Duration
+	if secondsString, ok := annotations[constants.ReconciliationFrequencyAnnotation]; ok {
+		seconds, err := strconv.Atoi(secondsString)
+		if err != nil {
+			l.Info("Failed to convert frequency annotation: defaulting to 120 seconds")
```
[the remark this answers is not in the document]
Comment (ADorigi) [reply pkg/plugins/plugin_frequency.go]: I've modified the log to include the error message.

**Comment by TylerGillson** [comment]
@ADorigi thanks for making those changes. We just need a `make reviewable` and it'll be good to go!

**Comment by ADorigi** [comment]
@TylerGillson 
Running `make reviewable` made a few changes in `chart/validator/values.yaml` and `chart/validator/README.md`, which may not be as expected. Let me know if these need to be reverted.

**Comment by TylerGillson** [comment]
@ADorigi you needed to do a `git submodule update --remote` then `make reviewable`. I did it for you though, so no worries this time.

**Review by TylerGillson — APPROVED** [review]
lgtm

**Pull request merged**